script.c

```diff
@@ -16,78 +16,90 @@
 
 #include "script.h"
 #include "plostd.h"
+#include "phfs.h"
 #include "cmd.h"
 #include "hal.h"
+#include "errors.h"
 
 
 /* Linker symbol points to the beginning of .data section */
 extern char script[];
 
 
 struct {
-	u32 cmdCnt;
-	char basicCmds[MAX_SCRIPT_LINES_NB][LINESZ];
+	u32 cnt;
+	char cmds[MAX_SCRIPT_LINES_NB][LINESZ];
 } script_common;
 
 
+static int script_regFile(char *cmd)
+{
+	int i;
+	size_t sz = 0;
+	addr_t addr = 0;
+	unsigned int pos = 0;
+	char alias[3][LINESZ + 1];
+
+	for (i = 0; i < 3; ++i) {
+		if (cmd_getnext(cmd, &pos, "(:) \t", NULL, alias[i], sizeof(alias[i])) == NULL || *alias[i] == 0)
+			break;
+
+		if (i == 1) {
+			if (plostd_ishex(alias[i]) < 0){
+				return ERR_ARG;
+			}
+			addr = plostd_ahtoi(alias[i]);
+		}
+		else if (i == 2) {
+			if (plostd_ishex(alias[i]) < 0) {
+				return ERR_ARG;
+			}
+
+			sz = plostd_ahtoi(alias[i]);
+
+		}
+	}
+
+	if (phfs_regFile(alias[0] + 1, addr, sz) < 0)
+		return ERR_ARG;
+
+	return ERR_NONE;
+}
+
+
 void script_init(void)
 {
 	int i;
 	unsigned int pos = 0;
 	char *argsScript = (char *)script;
 
-	script_common.cmdCnt = 0;
```

Comment:
Why are you removing this line?

Reply:
In this [**PR**](https://github.com/phoenix-rtos/plo/pull/54) the `.bss` section is zeroed so that this variable does not have to be initialized again in this function if we assume that `script_init()` is invoked only once.